features/field-based-acl.md

```diff
@@ -0,0 +1,190 @@
+Field-based ACL
+================
+
+# API Design Overview
+
+Field-based ACL is a Skygear feature defining the accessibility of the fields
+of a Skygear Record type.
+
+## Accessibility levels
+
+The field-based accessibility is divided into 5 levels:
+
+- `ReadWrite`: fields that available for read and write
+- `Queryable`: Fields that can be used to perform any query predicates and
+  will be returned after query.
+- `Discoverable`: Fields that can only be used to perform equality query
+  predicate and will be returned in query.
```

Comment:
I remember at some stage someone suggested that we can just make a discover API for this, would that proposal simpler? (a.k.a. Discoverable would meant developer can use the discover API to retrieve record, but not Query at all)

Otherwise I think a strict definition of Query like what proposed here is fine too.